app/src/components/page/Sidebar/utils.tsx

```diff
@@ -0,0 +1,24 @@
+import { Route } from '../../../routes';
+
+/**
+ *
+ * @param routes - The routes to get search the active route from
+ * @param pathname - an array of menu location path
+ */
+export function getActiveKey(pathname: string[], routes: Route[]) {
+  const url = '/' + pathname.join('/');
+  let activeKey: string | undefined;
+
+  function mapMenus(menu: Route) {
+    // Matching Url
+    if (menu.url && url.includes(menu.url)) activeKey = menu.key;
```

Comment:
do we need this check or is the one below sufficient `(url === menu.url)`?

Reply:
also maybe in the spirit of pure functions we extract the mapMenus function, have it take url as an arg and return the function instead of activeKey?

Reply:
e.g if we are on a route `/location/add` and in the sidebar we need to find an active key for it but in the sidebar we dont have any route for it that is why this condition will find the url least matching to it and that be `/inventory` thus we also have to find the matching url as well as the exact one if present

Reply:
> also maybe in the spirit of pure functions we extract the mapMenus function, have it take url as an arg and return the function instead of activeKey?

sorry i didnt got you 😅

Reply:
Extract the nested function to it's own standalone pure function